packages/ui/src/modal/buy/BuyModalRenderer.tsx

```diff
@@ -728,7 +730,7 @@ export const BuyModalRenderer: FC<Props> = ({
         blockExplorerBaseName,
         steps,
         stepData,
-        quantity,
+        quantity: quantity > quantityRemaining ? quantityRemaining : quantity,
```

Comment:
Would be best to have a useEffect that actually checks the quantity remaining whenever it changes and then ensures the quantity state will never be higher than the remaining quantity.